asv_bench/benchmarks/index_object.py

```diff
@@ -181,4 +181,16 @@ def time_get_loc(self):
         self.ind.get_loc(0)
 
 
+class IntervalIndexMethod(object):
+    # GH 24813
+    def setup(self):
+        N = 10**5
+        left = np.append(np.arange(N), np.array(0))
+        right = np.append(np.arange(1, N + 1), np.array(1))
+        self.intv = IntervalIndex.from_arrays(left, right)
+
+    def time_monotonic_inc(self):
+        self.intv.is_monotonic_increasing
```

Comment:
Just note that this measures access speed to the cached value, not the time cost of the underlying operation. To time the underlying method, you need to do ``pd.IntervalIndex.is_monotonic_increasing.func(self.intv)``, but that hits a cached value at ``self.intv._multiindex.is_monotonic_increasing`` in master, so will still seem to be in the nanosecond range, but really won't be because it's just the caching doing tricks.

Reply:
You're right. The performance is even better. The `_engine` is set in prior in asv setup method.
```
All benchmarks:

       before           after         ratio
     [a1fee919]       [6fb6d2cb]
     <master>         <GH24813-inc-dec>
          229±1ns          209±1ns     0.91  index_object.IntervalIndexMethod.time_monotonic_inc(1000)
-        450±10ns          260±2ns     0.58  index_object.IntervalIndexMethod.time_monotonic_inc(100000)
```